simulator/src/main/java/simblock/node/Farmer.java

```diff
@@ -0,0 +1,105 @@
+package simblock.node;
+
+import simblock.block.Block;
+import simblock.block.PoSpaceBlock;
+import simblock.simulator.Simulator;
+import simblock.simulator.Timer;
+import simblock.task.AbstractMintingTask;
+
+import java.math.BigInteger;
+
+import static simblock.settings.SimulationConfiguration.chia_k;
+import static simblock.simulator.Simulator.arriveBlock;
+
+public class Farmer extends Node{
+
+    private int[] PoSpaceCount;
+    private PoSpaceBlock[] Chain;
+    public Farmer(
+            int nodeID, int numConnection, int region, long miningPower, String routingTableName, boolean useCBR, boolean isChurnNode, int depth
+    ) {
+        super(nodeID, numConnection, region, miningPower, routingTableName, useCBR, isChurnNode);
+        PoSpaceCount = new int[depth];
```

Comment:
From the algorithms, how many times the Farmer has extended a block at the specified depth.
ie: PoSpaceCount[5] stores how many times a node has extended a block at depth 5.

Reply:
Resolving based on Tuesday discussion - unresolve if you don't think we sufficiently discussed this